Gems/PhysX/Code/Source/EditorMeshColliderComponent.cpp

```diff
@@ -0,0 +1,1101 @@
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &EditorProxyAssetShapeConfig::m_configuration,
+                        "Configuration",
+                        "PhysX mesh asset collider configuration.")
+                    ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &EditorProxyAssetShapeConfig::m_subdivisionLevel,
+                        "Subdivision level",
+                        "The level of subdivision if a primitive shape is replaced with a convex mesh due to scaling.")
+                    ->Attribute(AZ::Edit::Attributes::Min, Utils::MinCapsuleSubdivisionLevel)
+                    ->Attribute(AZ::Edit::Attributes::Max, Utils::MaxCapsuleSubdivisionLevel)
+                    ->Attribute(AZ::Edit::Attributes::Visibility, &EditorProxyAssetShapeConfig::ShowingSubdivisionLevel)
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorProxyAssetShapeConfig::OnConfigurationChanged);
+            }
+        }
+    }
+
+    EditorProxyAssetShapeConfig::EditorProxyAssetShapeConfig(
+        const Physics::PhysicsAssetShapeConfiguration& assetShapeConfiguration)
+        : m_configuration(assetShapeConfiguration)
+    {
+    }
+
+    bool EditorProxyAssetShapeConfig::ShowingSubdivisionLevel() const
+    {
+        return m_hasNonUniformScale;
+    }
+
+    AZ::u32 EditorProxyAssetShapeConfig::OnConfigurationChanged()
+    {
+        return AZ::Edit::PropertyRefreshLevels::ValuesOnly;
+    }
+
+    void EditorMeshColliderComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("PhysicsWorldBodyService"));
+        provided.push_back(AZ_CRC_CE("PhysicsColliderService"));
+        provided.push_back(AZ_CRC_CE("PhysicsTriggerService"));
+    }
+
+    void EditorMeshColliderComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("TransformService"));
+        required.push_back(AZ_CRC_CE("PhysicsRigidBodyService"));
+    }
+
+    void EditorMeshColliderComponent::GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    {
+        dependent.push_back(AZ_CRC_CE("NonUniformScaleService"));
+    }
+
+    void EditorMeshColliderComponent::Reflect(AZ::ReflectContext* context)
+    {
+        EditorProxyAssetShapeConfig::Reflect(context);
+
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<EditorMeshColliderComponent, EditorComponentBase>()
+                ->Version(1 + (1<<PX_PHYSICS_VERSION_MAJOR)) // Use PhysX version to trigger prefabs recompilation when switching between PhysX 4 and 5.
```

Comment:
@SergeyAMZN can speak to this, I believe the intention was just to have a big change to ensure the assets would get reprocessed and it intentionally didn't want just a single increment when switching back and forward

Reply:
Correct, this is to make it so the class has different versions when switching PhysX versions because cooked data can be added to the serialized data of the component, changing the version forces asset processor to reprocess the prefab again. But I thought I only needed it in the Mesh Collider, which is incorrect, the primitives colliders can also cooked data, for example for cylinders. I'll add it back to the `EditorColliderComponent` class too. I verified this with @SergeyAMZN.